src/mmda/utils/stringify.py

```diff
@@ -0,0 +1,128 @@
+"""
+
+Utility that converts a List[SpanGroup] into a string
+
+@kylel, @lucas
+
+Prior versions: 
+- https://github.com/allenai/timo_scim/blob/9fa19cd29cde0e2573e0079da8d776895f4d6caa/scim/predictors/utils.py#L221-L234
+- 
+
+"""
+
+import logging
+from typing import List, Optional
+
+from mmda.types import Document, Span, SpanGroup
+
+logging.basicConfig(level=logging.WARNING)
+logger = logging.getLogger(__name__)
+
+
+def stringify_span_group(
+    span_group: SpanGroup,
+    document: Document,
+    use_word_text: Optional[bool] = True,
+    replace_newlines_with: Optional[str] = " ",
+    join_words_with: Optional[str] = " ",
+    normalize_whitespace: Optional[bool] = True,
+    allow_partial_word_match: Optional[bool] = True,
+    allow_disjoint_spans: Optional[bool] = True,
+    include_symbols_between_disjoint_spans: Optional[bool] = False,
+) -> str:
+    """Requires doc.words to exist. This technique applies 4 main steps:
```

Comment:
where does `doc.words` come from?

Reply:
I may not have perfectly followed the conversation in the earlier SVM ticket/slack, but what is happening here as far as punctuation? We expect that infelicitous hyphens have been removed already in the words? What about the issue as far as, say, author names having undesirable commas clumped with them? How is that solved?

Reply:
I'm seeing some tests relevant to this, where the symbols include hyphens and line breaks that aren't in the words, and where serial commas end up as their own word. Is that the main resolution of the earlier question? The word predictor doesn't attempt to combine tokens with no whitespace between, but it does (we hope) successfully remove hyphens related to columns and pagination? And then it is up to the caller of stringify to have accurate enough span group boundaries to not get stray commas if they aren't wanted?

Reply:
Is there a way we can see if that actually works? I have trouble with this concept that we might be using the spans that come with the words to determine the stringify result, and that these are different from the spans the caller actually included in request